ipalib/__init__.py

```diff
@@ -942,12 +942,17 @@ def packages(self):
                 ipaserver.plugins,
             )
         else:
-            import ipaclient.remote_plugins
-            import ipaclient.plugins
-            result = (
-                ipaclient.remote_plugins.get_package(self),
-                ipaclient.plugins,
-            )
+            try:
+                import ipaclient.remote_plugins
+                import ipaclient.plugins
+                result = (
+                    ipaclient.remote_plugins.get_package(self),
+                    ipaclient.plugins,
+                )
+            except NetworkError:
+                #  instead of raising the default error connection message,
+                #  raising a more user-friendly one
+                raise GenericError('Cannot find IPA server to contact')
```

Comment:
That indeed is not a good idea since the api is finalized on client as well so it would most probably be always failing on clients.

I wonder whether we could always say that when `uri == http://localhost:8888/ipa/json`, the client is not installed. If so, there would perhaps be a way to do this check even earlier.
I think @MartinBasti might could give us pointer on the `uri` handling and whether `localhost` is special enough for us to treat it that way.